tests/api/api/test_artifacts.py

```diff
@@ -55,6 +56,52 @@ def _create_project(client: TestClient, project_name: str = PROJECT):
     return resp
 
 
+def test_store_artifact_with_invalid_key(db: Session, client: TestClient):
+    _create_project(client)
+    key_path = "some-key/with-slash"
+
+    resp = client.post(
+        STORE_API_ARTIFACTS_PATH.format(project=PROJECT, uid=UID, key=KEY, tag=TAG),
+        data="{}",
+    )
+    assert resp.status_code == HTTPStatus.OK.value
+
+    resp = client.post(
+        STORE_API_ARTIFACTS_PATH.format(
+            project=PROJECT, uid=UID, key=key_path, tag=TAG
+        ),
+        data="{}",
+    )
+    assert resp.status_code == HTTPStatus.BAD_REQUEST.value
+
+
+def test_store_artifact_backwards_compatibility(db: Session, client: TestClient):
+    _create_project(client)
+    key_path = "some-key/with-slash"
+    artifact = mlrun.artifacts.Artifact(key=KEY, body="123")
```

Comment:
I'd expect backwards compatibility test that
1. push invalid key (by mocking the regex?)
2. being able to pull it
3. being able to push to artifact another key that is valid
4. not being able to push to artifact another key that is invalid
5. being able to update it to a valid ? (is that supported at all? changing key? probably not)

Reply:
If I'm not mistaken, an artifact key cannot be changed once it's been created.
If you submit the request, it will generate another replica with the same key in the metadata, but only the 'db_key' field in the spec will contain the new key.
I also believe a change was about to be introduced regarding how you generate the artifact hash, which was supposed to change the way it works, I'll speak with Saar to clear this up and ensure how this works.

So 1is exactly what I did, and I'll add 2. and currently 3 - 5 is irrelevant, will update after talking to Saar.

Reply:
If I'm not mistaken, an artifact key cannot be changed once it's been created.
If you submit the request, it will generate another replica with the same key in the metadata, but only the 'db_key' field in the spec will contain the new key.
I also believe a change was about to be introduced regarding how you generate the artifact hash, which was supposed to change the way it works, I'll speak with Saar to clear this up and ensure how this works.

So 1is exactly what I did, and I'll add 2. and currently 3 - 5 is irrelevant, will update after talking to Saar.

Reply:
After clarifying with Saar, an artifact key really cannot be changed once it has been created, so 3-5 will create a new artifact in the database, making the check irrelevant. I'll add a check for pulling it to the test.